controllers/cloud.redhat.com/pool_controller.go

```diff
@@ -162,6 +208,12 @@ func (r *NamespacePoolReconciler) checkReadyNamespaceQuantity(pool crd.Namespace
 	size := pool.Spec.Size
 	ready := pool.Status.Ready
 	creating := pool.Status.Creating
+	reserved := pool.Status.Reserved
+
+	if ready+creating+reserved == pool.Spec.SizeLimit && pool.Spec.SizeLimit != 0 {
+		r.Log.Info(fmt.Sprintf("Max number of namespaces already created [%d] for pool [%s]", pool.Spec.SizeLimit, pool.Name))
+		return 0
```

Comment:
The function is called `checkReadyNamespaceQuantity`, yet we are returning 0, in a situation where we are out of spaces to create new ones, and later on, the number of available slots, though this also doesn't seem to take into account reserved. I think this should be checked.

Reply:
I think I resolved the ask here but let me know :smile: